core/unix/os.c

```diff
@@ -132,9 +132,15 @@ typedef struct rlimit64 rlimit64_t;
  */
 #define MCXT_SYSCALL_RES(mc) ((mc)->IF_X86_ELSE(xax, r0))
 #if defined(DR_HOST_AARCH64)
-#    define READ_TP_TO_R3_DISP_IN_R2    \
-        "mrs " ASM_R3 ", tpidr_el0\n\t" \
-        "ldr " ASM_R3 ", [" ASM_R3 ", " ASM_R2 "] \n\t"
+#    if defined(MACOS)
+#        define READ_TP_TO_R3_DISP_IN_R2      \
+            "mrs " ASM_R3 ", tpidrro_el0\n\t" \
```

Comment:
The only difference I see is `tpidrro_el0` instead of `tpidr_el0`.  It seems like that should be a change made in the  TLS headers with the usage here parameterized.